tests/kafkatest/tests/streams/streams_cooperative_rebalance_upgrade_test.py

```diff
@@ -0,0 +1,191 @@
+                                           1,
+                                           self.kafka,
+                                           self.source_topic,
+                                           throughput=1000,
+                                           acks=1)
+
+    @matrix(upgrade_from_version=streams_eager_rebalance_upgrade_versions)
+    def test_upgrade_to_cooperative_rebalance(self, upgrade_from_version):
+        self.zookeeper.start()
+        self.kafka.start()
+
+        processor1 = CooperativeRebalanceUpgradeService(self.test_context, self.kafka)
+        processor2 = CooperativeRebalanceUpgradeService(self.test_context, self.kafka)
+        processor3 = CooperativeRebalanceUpgradeService(self.test_context, self.kafka)
+
+        processors = [processor1, processor2, processor3]
+
+        # produce records continually during the test
+        self.producer.start()
+
+        # start all processors without upgrade_from config; normal operations mode
+        self.logger.info("Starting all streams clients in normal running mode")
+        for processor in processors:
+            processor.set_version(upgrade_from_version)
+            self.set_props(processor)
+            processor.CLEAN_NODE_ENABLED = False
+            # can't use state as older version don't have state listener
+            # so just verify up and running
+            verify_running(processor, self.processing_message)
+
+        # all running rebalancing has ceased
+        for processor in processors:
+            self.verify_processing(processor, self.processing_message)
+
+        # first rolling bounce with "upgrade.from" conifg set
+        previous_phase = ""
+        self.maybe_upgrade_rolling_bounce_and_verify(processors,
+                                                     previous_phase,
+                                                     self.first_bounce_phase,
+                                                     upgrade_from_version)
+
+        # All nodes processing, rebalancing has ceased
+        for processor in processors:
+            self.verify_processing(processor, self.first_bounce_phase + self.processing_message)
+
+        # second rolling bounce without "upgrade.from" conifg
+        self.maybe_upgrade_rolling_bounce_and_verify(processors,
+                                                     self.first_bounce_phase,
+                                                     self.second_bounce_phase)
+
+        # All nodes processing, rebalancing has ceased
+        for processor in processors:
+            self.verify_processing(processor, self.second_bounce_phase + self.processing_message)
+
+        # now verify tasks are unique
+        for processor in processors:
+            self.get_tasks_for_processor(processor)
+            self.logger.info("Active tasks %s" % processor.active_tasks)
+
+        overlapping_tasks = processor1.active_tasks.intersection(processor2.active_tasks, processor3.active_tasks)
```

Comment:
Why only testing intersect(1, 2, 3) and intersect(2, 3) here? Reading on https://www.geeksforgeeks.org/intersection-function-python/ I think if there's a task owned by both 1 and 2 but not 3, or by both 2 and 3 but not 1, then this would still pass?

Reply:
Ack, updated the tests to be 
```python
processor1.instersection(processor2)
processor1.instersection(processor3)
processor2.instersection(processor3)
```